Src/Support/Google.Apis.Auth/OAuth2/CredentialFactory.cs

```diff
@@ -57,60 +71,79 @@ public static T FromFile<T>(string credentialPath)
         return FromStream<T>(fileStream);
     }
 
+    /// <summary>
+    /// Creates a credential of the specified type from a file that contains JSON credential data.
+    /// </summary>
+    /// <param name="credentialPath">The path to the credential file.</param>
+    /// <param name="credentialType">The type of credential to be loaded. Valid strings can be found in <see cref="JsonCredentialParameters"/>.</param>
+    /// <returns>The created credential.</returns>
+    public static GoogleCredential FromFile(string credentialPath, string credentialType)
+    {
+        using FileStream fileStream = File.OpenRead(credentialPath);
+        return FromStream(fileStream, credentialType);
+    }
+
     /// <summary>
     /// Creates a credential of the specified type from a stream that contains JSON credential data.
     /// </summary>
     /// <typeparam name="T">The type of the credential to create.</typeparam>
     /// <param name="stream">The stream that contains the JSON credential data.</param>
     /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
     /// <returns>A task that will be completed with the created credential.</returns>
-    public static async Task<T> FromStreamAsync<T>(Stream stream, CancellationToken cancellationToken)
-    {
-        try
-        {
-            return FromJsonParameters<T>(await NewtonsoftJsonSerializer.Instance.DeserializeAsync<JsonCredentialParameters>(stream, cancellationToken).ConfigureAwait(false));
-        }
-        catch (Exception e)
-        {
-            throw new InvalidOperationException("Error deserializing JSON credential data.", e);
-        }
-    }
+    public static async Task<T> FromStreamAsync<T>(Stream stream, CancellationToken cancellationToken) =>
+        FromJsonParameters<T>(await HandleDeserializationErrorsAsync(() =>
+            NewtonsoftJsonSerializer.Instance.DeserializeAsync<JsonCredentialParameters>(stream, cancellationToken)).ConfigureAwait(false));
+
+    /// <summary>
+    /// Creates a credential of the specified type from a stream that contains JSON credential data.
+    /// </summary>
+    /// <param name="stream">The stream that contains the JSON credential data.</param>
+    /// <param name="credentialType">The type of credential to be loaded. Valid strings can be found in <see cref="JsonCredentialParameters"/>.</param>
+    /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+    /// <returns>A task that will be completed with the created credential.</returns>
+    public static async Task<GoogleCredential> FromStreamAsync(Stream stream, string credentialType, CancellationToken cancellationToken) =>
```

Comment:
I'm very tempted to keep these private. We only need the FromFile ones for Spanner et all.